emission/tests/modellingTests/TestSimilarityMetric.py

```diff
@@ -6,26 +6,93 @@ class TestSimilarityMetric(unittest.TestCase):
 
     def testODsAreSimilar(self):
         generate_points_thresh = 0.001  # approx. 111 meters
-        similarity_threshold = 500  # 
-        # random, but, points are sampled within a circle and should always be < sim threshold
-        trips = etmm.generate_mock_trips('bob', 2, [0, 0], [1, 1], threshold=generate_points_thresh)
+        similarity_threshold = 111  # 
+
         metric = eamso.OriginDestinationSimilarity()
+        ## Sub-Test 1 - 3 :
+        # random, but, origin and destination points are sampled within a circle and should always be < sim threshold
+        # Since both origin and destination poitns lie within threshold limits,they should be similar
+        # when we check by just origin or just destination or both origin-and-destination
+
+        trips = etmm.generate_mock_trips('bob', 2, [0, 0], [1, 1], 'od',threshold=generate_points_thresh) 
         coords0 = metric.extract_features(trips[0])
-        coords1 = metric.extract_features(trips[1])
-        similar = metric.similar(coords0, coords1, similarity_threshold)
-        self.assertTrue(similar)
+        coords1 = metric.extract_features(trips[1])        
+        similarOD1 = metric.similar(coords0, coords1, similarity_threshold)
+        similarOD2 = metric.similar(coords0[:2], coords1[:2], similarity_threshold)
+        similarOD3 = metric.similar(coords0[2:], coords1[2:], similarity_threshold)
+
+        ## Sub-Test 4 :
+        # random, but, only origin points are sampled within a circle and should always be < sim threshold
+        # Since origin of two points lies within threshold limits,they should be similar
+        # when we check just origin for similarity.
+
+
+        trips = etmm.generate_mock_trips('alice', 2, [0, 0], [1, 1], 'o_',threshold=generate_points_thresh)        
+        coords0 = metric.extract_features(trips[0])[:2]
+        coords1 = metric.extract_features(trips[1])[:2]        
+        similarO = metric.similar(coords0, coords1, similarity_threshold)
+
+        ##Sub-Test 5 :
+        # random, but, only destination points are sampled within a circle and should always be < sim threshold
+        # Since destination of two points lies within threshold limits,they should be similar
+        # when we check just destination for similarity.
+
+        trips = etmm.generate_mock_trips('Caty', 2, [0, 0], [1, 1], '_d',threshold=generate_points_thresh)        
+        coords0 = metric.extract_features(trips[0])[2:]
+        coords1 = metric.extract_features(trips[1])[2:]        
+        similarD = metric.similar(coords0, coords1, similarity_threshold)
+
+        # All the similars must be true
+        self.assertTrue(similarOD1) # RESULT SUB-TEST 1
+        self.assertTrue(similarOD2) # RESULT SUB-TEST 2
+        self.assertTrue(similarOD3) # RESULT SUB-TEST 3
+        self.assertTrue(similarO)  # RESULT SUB-TEST 4
+        self.assertTrue(similarD) # RESULT SUB-TEST 5
```

Comment:
I would put the assert right after the test so that it is clear what you are testing